### PR TITLE
[ONNX importer] Add support for Usample-8 and Upsample-9

### PR DESCRIPTION
**Pull request by mitruska**
Support for [Usample-8](https://github.com/onnx/onnx/blob/master/docs/Changelog.md#upsample-7) (or lower) and [Upsample-9](https://github.com/onnx/onnx/blob/master/docs/Changelog.md#upsample-9) in ONNX importer.
The difference is that in Upsample-9 `scales` were moved from attributes to inputs.

### CODE REVIEW

File: ngraph/src/ngraph/frontend/onnx_import/op/upsample.cpp [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+                    const auto mode = node.get_attribute_value<std::string>("mode", "nearest");
+                    check_mode_support(node, mode);
+
+                    size_t axes_size = scales.size();
+
+                    AxisSet axes;
+                    for (int ax = 0; ax < axes_size; ++ax)
+                    {
+                        axes.insert(ax);
+                    }
```
Comment (postrational) [inline ngraph/src/ngraph/frontend/onnx_import/op/upsample.cpp:75]: Consider using `std::iota` here...
```suggestion
    AxisSet axes(axes_size);
    std::iota(axes.begin(), axes.end(), 0);
```

File: ngraph/src/ngraph/frontend/onnx_import/op/upsample.cpp [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+                    CHECK_VALID_NODE(
+                        node,
+                        (scales_shape.is_static() || data_shape.rank().is_static()),
+                        " Data rank or shape of Scales input is required to be static.");
+
+                    size_t axes_size = scales_shape.is_static() ? scales_shape.to_shape().at(0)
+                                                                : data_shape.rank().get_length();
+                    AxisSet axes;
+                    for (int ax = 0; ax < axes_size; ++ax)
```
Comment (postrational) [inline ngraph/src/ngraph/frontend/onnx_import/op/upsample.cpp:136]: `std::iota`?

File: ngraph/src/ngraph/frontend/onnx_import/op/upsample.cpp (hunk 1 above)
Comment (mitruska) [reply ngraph/src/ngraph/frontend/onnx_import/op/upsample.cpp:75]: Sure, I thought about it, but I wasn't sure is that makes sense for `unordered_set`.

File: ngraph/test/onnx/onnx_import.in.cpp [hunk 3]
```diff
@@ -2105,3 +2105,105 @@ NGRAPH_TEST(${BACKEND_NAME}, onnx_model_scatter_elements_import_only)
...
+
+NGRAPH_TEST(${BACKEND_NAME}, onnx_upsample8_import_only)
+{
+    const auto function = onnx_import::import_onnx_model(
+        file_util::path_join(SERIALIZED_ZOO, "onnx/upsample8_nearest.prototxt"));
+
+    // Input data shape (1, 1, 2, 2)
+    // Scales attribute values {1.0, 1.0, 2.0, 3.0}
+
+    Shape expected_output_shape{1, 1, 4, 6};
```
Comment (tomdol) [inline ngraph/test/onnx/onnx_import.in.cpp:2117]:
```suggestion
    const Shape expected_output_shape{1, 1, 4, 6};
```

File: ngraph/test/onnx/onnx_import.in.cpp [hunk 4]
```diff
@@ -2105,3 +2105,105 @@ NGRAPH_TEST(${BACKEND_NAME}, onnx_model_scatter_elements_import_only)
...
+NGRAPH_TEST(${BACKEND_NAME}, onnx_upsample8_nearest_infer)
+{
+    const auto function = onnx_import::import_onnx_model(
+        file_util::path_join(SERIALIZED_ZOO, "onnx/upsample8_nearest.prototxt"));
+
+    // Input data shape (1, 1, 2, 2)
+    // Scales attribute values {1.0, 1.0, 2.0, 3.0}
+    // mode: nearest
+
+    Shape expected_output_shape{1, 1, 4, 6};
```
Comment (tomdol) [inline ngraph/test/onnx/onnx_import.in.cpp:2133]:
```suggestion
    const Shape expected_output_shape{1, 1, 4, 6};
```

File: ngraph/test/onnx/onnx_import.in.cpp [hunk 5]
```diff
@@ -2105,3 +2105,105 @@ NGRAPH_TEST(${BACKEND_NAME}, onnx_model_scatter_elements_import_only)
...
+NGRAPH_TEST(${BACKEND_NAME}, onnx_upsample8_linear_infer)
+{
+    const auto function = onnx_import::import_onnx_model(
+        file_util::path_join(SERIALIZED_ZOO, "onnx/upsample8_linear.prototxt"));
+
+    // Input data shape (1, 1, 2, 2)
+    // Scales attribute values {1.0, 1.0, 2.0, 2.0}
+    // mode: linear
+
+    Shape expected_output_shape{1, 1, 4, 4};
```
Comment (tomdol) [inline ngraph/test/onnx/onnx_import.in.cpp:2151]:
```suggestion
    const Shape expected_output_shape{1, 1, 4, 4};
```

File: ngraph/test/onnx/onnx_import.in.cpp [hunk 6]
```diff
@@ -2105,3 +2105,105 @@ NGRAPH_TEST(${BACKEND_NAME}, onnx_model_scatter_elements_import_only)
...
+
+NGRAPH_TEST(${BACKEND_NAME}, onnx_upsample9_scales_const_import_only)
+{
+    const auto function = onnx_import::import_onnx_model(
+        file_util::path_join(SERIALIZED_ZOO, "onnx/upsample9_scales_const_nearest.prototxt"));
+
+    // Input data shape (1, 1, 2, 2)
+    // Input const scales values {1.0, 1.0, 2.0, 3.0}
+
+    Shape expected_output_shape{1, 1, 4, 6};
```
Comment (tomdol) [inline ngraph/test/onnx/onnx_import.in.cpp:2168]:
```suggestion
    const Shape expected_output_shape{1, 1, 4, 6};
```

File: ngraph/test/onnx/onnx_import.in.cpp [hunk 7]
```diff
@@ -2105,3 +2105,105 @@ NGRAPH_TEST(${BACKEND_NAME}, onnx_model_scatter_elements_import_only)
...
+NGRAPH_TEST(${BACKEND_NAME}, onnx_upsample9_scales_const_nearest_infer)
+{
+    const auto function = onnx_import::import_onnx_model(
+        file_util::path_join(SERIALIZED_ZOO, "onnx/upsample9_scales_const_nearest.prototxt"));
+
+    // Input data shape (1, 1, 2, 2)
+    // Input const scales values {1.0, 1.0, 2.0, 3.0}
+    // mode: nearest
+
+    Shape expected_output_shape{1, 1, 4, 6};
```
Comment (tomdol) [inline ngraph/test/onnx/onnx_import.in.cpp:2184]:
```suggestion
    const Shape expected_output_shape{1, 1, 4, 6};
```

File: ngraph/test/onnx/onnx_import.in.cpp [hunk 8]
```diff
@@ -2105,3 +2105,105 @@ NGRAPH_TEST(${BACKEND_NAME}, onnx_model_scatter_elements_import_only)
...
+NGRAPH_TEST(${BACKEND_NAME}, onnx_upsample9_scales_const_linear_infer)
+{
+    const auto function = onnx_import::import_onnx_model(
+        file_util::path_join(SERIALIZED_ZOO, "onnx/upsample9_scales_const_linear.prototxt"));
+
+    // Input data shape (1, 1, 2, 2)
+    // Input const scales values {1.0, 1.0, 2.0, 2.0}
+    // mode: linear
+
+    Shape expected_output_shape{1, 1, 4, 4};
```
Comment (tomdol) [inline ngraph/test/onnx/onnx_import.in.cpp:2202]:
```suggestion
    const Shape expected_output_shape{1, 1, 4, 4};
```

File: ngraph/test/onnx/onnx_import_dyn_shapes.in.cpp [hunk 9]
```diff
@@ -1177,3 +1177,22 @@ NGRAPH_TEST(${BACKEND_NAME}, onnx_model_range_negative_step)
...
+
+NGRAPH_TEST(${BACKEND_NAME}, onnx_upsample9_scales_input_nearest_infer)
+{
+    const auto function = onnx_import::import_onnx_model(
+        file_util::path_join(SERIALIZED_ZOO, "onnx/upsample9_scales_input_nearest.prototxt"));
+
+    // Input data shape (1, 1, 2, 2)
+    // mode: nearest
+
+    Shape expected_output_shape{1, 1, 4, 6};
```
Comment (tomdol) [inline ngraph/test/onnx/onnx_import_dyn_shapes.in.cpp:1189]:
```suggestion
    const Shape expected_output_shape{1, 1, 4, 6};
```

File: ngraph/src/ngraph/frontend/onnx_import/op/upsample.cpp [hunk 10]
```diff
@@ -0,0 +1,183 @@
...
+{
+    namespace onnx_import
+    {
+        namespace op
+        {
+            namespace
+            {
+                bool check_mode_support(const onnx_import::Node& node, const std::string mode)
+                {
+                    std::unordered_set<std::string> supported_modes = {"nearest", "linear"};
```
Comment (tomdol) [inline ngraph/src/ngraph/frontend/onnx_import/op/upsample.cpp:33]:
```suggestion
                    static const std::unordered_set<std::string> supported_modes = {"nearest", "linear"};
```

File: ngraph/src/ngraph/frontend/onnx_import/op/upsample.cpp (hunk 1 above)
Comment (mitruska) [reply ngraph/src/ngraph/frontend/onnx_import/op/upsample.cpp:75]: Actually, to use `std::iota` in this case we would need `std::vector` help. 
`AxisSet` inherits from `std::set` so it doesn't even have constructor that take size like you proposed or reserve function.

File: ngraph/src/ngraph/frontend/onnx_import/op/upsample.cpp (hunk 10 above)
Comment (mitruska) [reply ngraph/src/ngraph/frontend/onnx_import/op/upsample.cpp:33]: I am ok with `const`, but unconvinced that we really want `static` here.

File: ngraph/src/ngraph/frontend/onnx_import/op/upsample.cpp [hunk 11]
```diff
@@ -0,0 +1,177 @@
...
+
+                    const auto scales = node.get_attribute_value<std::vector<float>>("scales");
+                    const auto mode = node.get_attribute_value<std::string>("mode", "nearest");
+                    check_mode_support(node, mode);
+
+                    auto attrs = ngraph::op::v0::InterpolateAttrs();
+                    attrs.mode = mode;
+                    attrs.align_corners = false;
+
+                    for (size_t ax = 0; ax < scales.size(); ++ax)
```
Comment (arogowie-intel) [inline ngraph/src/ngraph/frontend/onnx_import/op/upsample.cpp:73]: What if scales size is less or greater than data rank?

File: ngraph/src/ngraph/frontend/onnx_import/op/upsample.cpp [hunk 12]
```diff
@@ -0,0 +1,177 @@
...
+                    CHECK_VALID_NODE(
+                        node,
+                        (scales_shape.is_static() || data_shape.rank().is_static()),
+                        " Data rank or shape of Scales input is required to be static.");
+
+                    auto attrs = ngraph::op::v0::InterpolateAttrs();
+                    attrs.mode = mode;
+                    attrs.align_corners = false;
+
+                    size_t axes_size = scales_shape.is_static() ? scales_shape.to_shape().at(0)
```
Comment (arogowie-intel) [inline ngraph/src/ngraph/frontend/onnx_import/op/upsample.cpp:133]: "_The number of elements of 'scales' should be the same as the rank of input 'X'._" I think you should enforce this condition.

**Pull request merged**